lib/matplotlib/docstring.py

```diff
@@ -0,0 +1,5 @@
+from matplotlib._docstring import *  # noqa: F401, F403
+from matplotlib import _api
+_api.warn_deprecated(
+    "3.6", message="The module %(name)s is deprecated since %(since)s.",
```

Comment:
Do you need to pass `message` explicitly instead of passing `obj_type='module'`?

Reply:
Maybe not. This is taken from #16181 (although with an added end of the message). I'll dig into it.

Reply:
With this:

```
In [1]: from matplotlib import docstring
/tmp/ipykernel_1388/3995544149.py:1: MatplotlibDeprecationWarning: The module matplotlib.docstring is deprecated since 3.6.
  from matplotlib import docstring
```

With `obj_type='module'`:
```
In [1]: from matplotlib import docstring
/tmp/ipykernel_4063/3995544149.py:1: MatplotlibDeprecationWarning: 
The matplotlib.docstring module was deprecated in Matplotlib 3.6 and will be removed two minor releases later.
  from matplotlib import docstring
```

Reply:
I went with your suggestion.

Reply:
It doesn't seem like it was changed, though?

Reply:
Sorry. Been editing on multiple computers and must have missed pulling before editing at some stage.

Reply:
You may want to use `git push --force-with-lease` instead of `git push --force`, if you are working on multiple computers.

Reply:
Yeah, except that my office computer is running CentOS 7, so git is too old to have --force-with-lease...